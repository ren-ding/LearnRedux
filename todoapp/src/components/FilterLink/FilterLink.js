import React from 'react';
import {connect} from 'react-redux'

const Link = ({
    active,
    children,
    onClick
}) => {
    if(active){
        return <span>{children}</span>;
    }
    return(
    <a href='#'
       onClick={ e=>{
            e.preventDefault();
            onClick();
        }}
    >
    {children}
    </a>
    );
};

const mapStateToLinkProps = (
    state,
    ownProps
) => {
    return {
        active: ownProps.filter === state.visibilityFilter
    }
}

const mapDispatchToProps = (
    dispatch,
    ownProps
) => {
    return {
        onClick: () => {
            dispatch({
                type: 'SET_VISIBILITY_FILTER',
                filter: ownProps.filter
            });
        }
    }
}

const FilterLink = connect(
    mapStateToLinkProps,
    mapDispatchToProps
)(Link);


export default FilterLink;