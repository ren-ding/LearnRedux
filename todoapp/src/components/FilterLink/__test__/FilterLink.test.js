import React from 'react';
import renderer from 'react-test-renderer';
import FilterLink from '../FilterLink';
import {BrowserRouter} from 'react-router-dom';

describe('FilterLink',()=>{ 
    let props;

    beforeEach(()=>{
        props = {
            filter: 'all'
        }
    });

    describe('render',()=>{
        it('should render match with snapshot',()=> {
            const tree = renderer.create(
                <BrowserRouter>
                <FilterLink {...props}/>
                </BrowserRouter>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});