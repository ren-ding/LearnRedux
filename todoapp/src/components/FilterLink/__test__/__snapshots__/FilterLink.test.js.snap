// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLink render should render match with snapshot 1`] = `
<a
  aria-current="page"
  className="active"
  href="/"
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "textDecoration": "none",
    }
  }
/>
`;
